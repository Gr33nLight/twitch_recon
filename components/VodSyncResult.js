import React from "react";
import { Link, Box, Icon, Text } from "@chakra-ui/react";
import { FaTrashAlt } from "react-icons/fa";

const VodSyncResult = ({ result, setVodResult, setVodUrl }) => {
  return (
    <Box
      padding="10px"
      bgColor="purple.400"
      width="300px"
      display="inherit"
      marginTop={["5px", "0px"]}
    >
      <Text fontSize="md" color="gray.200" width="100%">
        <Link href={result} isExternal>
          {result.substr(0, 31) + (result.length > 31 ? "..." : "")}
        </Link>
      </Text>
      <Icon
        cursor="pointer"
        onClick={() => {
          setVodResult("");
          setVodUrl("");
        }}
        w="5"
        h="5"
        as={FaTrashAlt}
        color="gray.200"
        mt="2px"
        ml="10px"
      />
    </Box>
  );
};

export default VodSyncResult;
