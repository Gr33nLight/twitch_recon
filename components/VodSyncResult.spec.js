import React from 'react';
import renderer from 'react-test-renderer';
import VodSyncResult from './VodSyncResult';

describe('<VodSyncResult />', () => {
  it('renders correctly', async () => {
    const tree = renderer
      .create(<VodSyncResult result={'random result'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
