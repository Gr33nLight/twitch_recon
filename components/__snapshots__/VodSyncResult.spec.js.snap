// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VodSyncResult /> renders correctly 1`] = `
<div
  className="css-zjarua"
>
  <p
    className="chakra-text css-ao67d5"
  >
    <a
      className="chakra-link css-0"
      href="random result"
      rel="noopener noreferrer"
      target="_blank"
    >
      random result
    </a>
  </p>
  <svg
    className="chakra-icon css-xbrkfx"
    fill="currentColor"
    focusable={false}
    height="1em"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
    />
  </svg>
</div>
`;
